libraries/AP_AHRS/AP_AHRS_DCM.cpp

```diff
@@ -440,14 +440,15 @@ AP_AHRS_DCM::drift_correction_yaw(void)
         /*
           we are using compass for yaw
          */
-        if (_compass->last_update_usec() != _compass_last_update) {
-            yaw_deltat = (_compass->last_update_usec() - _compass_last_update) * 1.0e-6f;
-            _compass_last_update = _compass->last_update_usec();
+        Compass &_compass = AP::compass();
```

Comment:
put this sooner ?

Reply:
It's scoped as tightly as is sensible IMO.... it's not needed my most of the code in the function.